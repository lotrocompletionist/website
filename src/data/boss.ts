export interface IBoss {
  id: number;
  name: string;
}
