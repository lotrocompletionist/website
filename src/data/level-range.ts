export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
