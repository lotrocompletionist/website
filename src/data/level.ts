export interface Level {
  minimum: number;
  maximum?: number;
}
