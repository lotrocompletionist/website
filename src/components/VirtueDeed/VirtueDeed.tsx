import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { IVirtueDeed } from "../../data/virtue-deed";

export interface VirtueDeedProps {
  virtueDeed: IVirtueDeed;
}

export function VirtueDeed(props: VirtueDeedProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.virtueDeed.deed}</Title>
          </MediaContent>
        </Media>
        <Content>
          {props.virtueDeed.virtue}
          <br />
          <small>
            {props.virtueDeed.level}
            <br/>
            {props.virtueDeed.region}
          </small>
        </Content>
      </CardContent>
    </>
  );
}
