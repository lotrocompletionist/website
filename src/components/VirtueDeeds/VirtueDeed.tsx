import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { IVirtueDeed } from "@app/data/virtue-deeds";
import { Level } from "@app/components/Shared/Level/Level";

export interface VirtueDeedProps {
  virtueDeed: IVirtueDeed;
}

export function VirtueDeed(props: VirtueDeedProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.virtueDeed.deed}</Title>
            <Subtitle isSize={6}>{props.virtueDeed.region}</Subtitle>
          </MediaContent>
        </Media>
        <Content>
          {props.virtueDeed.virtue}
          <br />
          <small>
            <Level level={props.virtueDeed.level} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
