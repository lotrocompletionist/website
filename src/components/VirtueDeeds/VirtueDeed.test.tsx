import * as React from "react";
import { create } from "react-test-renderer";
import { VirtueDeed } from "./VirtueDeed";
import { IVirtueDeed } from "@app/data/virtue-deed";

test("<VirtueDeed> renders correctly", () => {
  const virtueDeed: IVirtueDeed = {
    id: 1,
    virtue: "Charity",
    deed: "Bree-land Adventurer (Advanced)",
    region: "Bree-land Deeds",
    level: 4
  };
  const tree = create(<VirtueDeed virtueDeed={virtueDeed} />).toJSON();
  expect(tree).toMatchSnapshot();
});
