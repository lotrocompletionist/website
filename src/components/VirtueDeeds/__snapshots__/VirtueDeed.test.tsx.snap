// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VirtueDeed> renders correctly 1`] = `
<div
  className="card-content"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <figure
        className="image is-48x48"
      >
        <img
          src="https://via.placeholder.com/96x96"
        />
      </figure>
    </div>
    <div
      className="media-content"
    >
      <h1
        className="title is-4"
      >
        Bree-land Adventurer (Advanced)
      </h1>
      <h2
        className="subtitle is-6"
      >
        Bree-land Deeds
      </h2>
    </div>
  </article>
  <div
    className="content"
  >
    Charity
    <br />
    <small>
      Level: 
      4
    </small>
  </div>
</div>
`;
